contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/azkaban/VoldemortBuildAndPushJob.java

```diff
@@ -816,6 +822,20 @@ public void runPushStore(Props props, String url, String dataDir) throws Excepti
 
         log.info("Push starting for cluster: " + url);
 
+        // CDN distcp
+        boolean cdnEnabled = props.getBoolean(VoldemortBuildAndPushJob.PUSH_CDN_ENABLED, false);
+        String modifiedDataDir = new Path(dataDir).makeQualified(FileSystem.get(new JobConf())).toString();
+
+        if (cdnEnabled) {
+            if (modifiedDataDir.matches(".*hdfs://.*:[0-9]{1,5}/.*")) {
```

Comment:
So we mandate `hdfs` as the scheme? No other schemes are supported?

Can you move this validation code into `HadoopUtils` with a self-explanatory function name? I don't like to see regexes lying around in the code. It makes it harder to maintain.